src/coreclr/jit/gentree.cpp

```diff
@@ -26918,6 +26918,27 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad(GenTree** pAddr) const
             case NI_Sve_Load4xVectorAndUnzip:
                 addr = Op(2);
                 break;
+
+            case NI_Sve_GatherVector:
+            case NI_Sve_GatherVectorByteZeroExtend:
+            case NI_Sve_GatherVectorInt16SignExtend:
+            case NI_Sve_GatherVectorInt16WithByteOffsetsSignExtend:
+            case NI_Sve_GatherVectorInt32SignExtend:
+            case NI_Sve_GatherVectorInt32WithByteOffsetsSignExtend:
+            case NI_Sve_GatherVectorSByteSignExtend:
+            case NI_Sve_GatherVectorUInt16WithByteOffsetsZeroExtend:
+            case NI_Sve_GatherVectorUInt16ZeroExtend:
+            case NI_Sve_GatherVectorUInt32WithByteOffsetsZeroExtend:
+            case NI_Sve_GatherVectorUInt32ZeroExtend:
+                addr = Op(2);
+                if (varTypeIsSIMD(addr->gtType))
+                {
+                    // The address is a vector of addresses.
+                    // Return true, but do not set pAddr.
```

Comment:
@a74nh  - do you remember why we do not set `pAddr` here? We get AV in value numbering because of that. 

`Sve_GatherVector_Bases_double_ulong` was failing for me with below call stack:

```
>	clrjit_universal_arm64_x64.dll!Compiler::fgValueNumberAddExceptionSetForIndirection(GenTree * tree, GenTree * baseAddr) Line 13618	C++
 	clrjit_universal_arm64_x64.dll!Compiler::fgValueNumberHWIntrinsic(GenTreeHWIntrinsic * tree) Line 12461	C++
 	clrjit_universal_arm64_x64.dll!Compiler::fgValueNumberTree(GenTree * tree) Line 12056	C++
 	clrjit_universal_arm64_x64.dll!Compiler::fgValueNumberBlock(BasicBlock * blk) Line 10554	C++
 	clrjit_universal_arm64_x64.dll!Compiler::fgValueNumber() Line 10340	C++
 	clrjit_universal_arm64_x64.dll!CompilerPhaseWithStatus::DoPhase() Line 132	C++
```

Reply:
Opened https://github.com/dotnet/runtime/issues/103606